core/templates/route_map.tpl

```diff
@@ -64,41 +64,47 @@ if(isset($schedule))
 /*	Below here is all the javascript for the map. Be careful of what you
 	modify!! */
 ?>
+<script src="<?php echo SITE_URL?>/lib/js/base_map.js"></script>
 <script type="text/javascript">
-var options = {
-	mapTypeId: google.maps.MapTypeId.ROADMAP
-}
 
-var map = new google.maps.Map(document.getElementById("routemap"), options);
-var dep_location = new google.maps.LatLng(<?php echo $mapdata->deplat?>,<?php echo $mapdata->deplng;?>);
-var arr_location = new google.maps.LatLng(<?php echo $mapdata->arrlat?>,<?php echo $mapdata->arrlng;?>);
+const map = createMap({
+	render_elem: 'routemap',
+});
```

Comment:
You can also pass in the base map provider here:

```js
provider: 'OpenStreetMap.Mapnik',
```

The default is `OpenStreetMap.Mapnik`